packages/framework-provider-azure/src/library/health-adapter.ts

```diff
@@ -0,0 +1,121 @@
+import { BoosterConfig, HealthEnvelope } from '@boostercloud/framework-types'
+import { Container, CosmosClient, PartitionKey } from '@azure/cosmos'
+import { environmentVarNames } from '../constants'
+import { Context } from '@azure/functions'
+import { request } from '@boostercloud/framework-common-helpers'
+
+export async function databaseUrl(cosmosDb: CosmosClient, config: BoosterConfig): Promise<Array<string>> {
+  const database = cosmosDb.database(config.resourceNames.applicationStack)
+  return [database.url]
+}
+
+export function getContainer(cosmosDb: CosmosClient, config: BoosterConfig, catalog: string): Container {
+  return cosmosDb.database(config.resourceNames.applicationStack).container(catalog)
+}
+
+export async function getConstant(
+  cosmosDb: CosmosClient,
+  config: BoosterConfig,
+  catalog: string
+): Promise<string | undefined> {
+  const container = getContainer(cosmosDb, config, catalog)
+  const { resources } = await container.items.query('SELECT 1 FROM c', { maxItemCount: -1 }).fetchAll()
+  return resources && resources.length > 0 ? resources[0] : undefined
+}
+
+export async function partitionKeys(container: Container): Promise<PartitionKey | undefined> {
+  const { resource } = await container.readPartitionKeyDefinition()
+  return resource
+}
+
+export async function countAll(container: Container): Promise<number> {
+  const { resources } = await container.items.query('SELECT VALUE COUNT(1) FROM c', { maxItemCount: -1 }).fetchAll()
+  return resources ? resources[0] : 0
+}
+
+export async function eventsCatalogHealthDetails(cosmosDb: CosmosClient, config: BoosterConfig): Promise<unknown> {
+  const container = getContainer(cosmosDb, config, config.resourceNames.eventsStore)
+  const url = container.url
+  const partitionKeyDefinition = await partitionKeys(container)
+  const count = await countAll(container)
+  return {
+    url: url,
+    partitionKeyDefinition: partitionKeyDefinition,
+    count: count,
```

Comment:
Wow, we return the count of events as part of the health check. This is one of the most detailed health check I've seen!